src/simtools/production_configuration/scale_events_manager.py

```diff
@@ -0,0 +1,117 @@
+
+    This class handles the evaluation of statistical uncertainties from DL2 MC event files
+    and performs interpolation for a specified query point.
+    """
+
+    def __init__(self, args_dict):
+        """
+        Initialize the manager with the provided arguments.
+
+        Parameters
+        ----------
+        args_dict : dict
+            Dictionary of command-line arguments.
+        """
+        self.args = args_dict
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(logging.INFO)
+        self.output_path = io_handler.IOHandler().get_output_directory(Path(__file__).stem)
+        self.output_filepath = Path(self.output_path).joinpath(f"{self.args['output_file']}")
+        self.metrics = collect_data_from_file(self.args["metrics_file"])
+        self.evaluator_instances = []
+
+    def initialize_evaluators(self):
+        """Initialize StatisticalErrorEvaluator instances for the given zeniths and offsets."""
+        if not (self.args["base_path"] and self.args["zeniths"] and self.args["offsets"]):
+            self.logger.warning("No files read")
+            self.logger.warning(f"Base Path: {self.args['base_path']}")
+            self.logger.warning(f"Zeniths: {self.args['zeniths']}")
+            self.logger.warning(f"Offsets: {self.args['offsets']}")
+            return
+
+        for zenith, offset in itertools.product(self.args["zeniths"], self.args["offsets"]):
+            file_name = self.args["file_name_template"].format(zenith=int(zenith))
+            file_path = Path(self.args["base_path"]).joinpath(file_name)
+
+            if not file_path.exists():
+                self.logger.warning(f"File not found: {file_path}. Skipping.")
+                continue
+
+            evaluator = StatisticalErrorEvaluator(
+                file_path,
+                file_type="Gamma-cone",
+                metrics=self.metrics,
+                grid_point=(1 * u.TeV, 180 * u.deg, zenith, 0, offset * u.deg),
+            )
+            evaluator.calculate_metrics()
+            self.evaluator_instances.append(evaluator)
+
+    def perform_interpolation(self):
+        """Perform interpolation for the query point."""
+        if not self.evaluator_instances:
+            self.logger.error("No evaluators initialized. Cannot perform interpolation.")
+            return None
+
+        interpolation_handler = InterpolationHandler(self.evaluator_instances, metrics=self.metrics)
+        query_point = self.args.get("query_point")
+        if not query_point or len(query_point) != 5:
+            raise ValueError(
+                "Invalid query point format. "
+                f"Expected 5 values, got {len(query_point) if query_point else 'None'}."
+            )
+        query_points = np.array([self.args["query_point"]])
+        return interpolation_handler.interpolate(query_points)
+
+    def write_output(self, scaled_events):
+        """Write the interpolation results to a file."""
```

Comment:
Be more concrete, what are 'results'? These are the derived / interpolated event statistics?

Reply:
fixed